logstash-core/src/main/java/org/logstash/ackedqueue/io/FileCheckpointIO.java

```diff
@@ -64,10 +63,12 @@ public Checkpoint write(String fileName, int pageNum, int firstUnackedPageNum, l
 
     @Override
     public void write(String fileName, Checkpoint checkpoint) throws IOException {
-        Path path = Paths.get(dirPath, fileName);
-        final byte[] buffer = new byte[BUFFER_SIZE];
         write(checkpoint, buffer);
-        Files.write(path, buffer, WRITE_OPTIONS);
+        buffer.flip();
+        try (FileOutputStream out = new FileOutputStream(Paths.get(dirPath, fileName).toFile())) {
```

Comment:
@jordansissel If I use `Files.write()` I have to operate in the on heap `byte[]` paradigm which forces another memory copy on me. This way I can go straight for writing out the buffer and retain the resilience by calling `sync`.